src/overlays/actors/ovl_En_Ice_Hono/z_en_ice_hono.c

```diff
@@ -21,37 +32,347 @@ const ActorInit En_Ice_Hono_InitVars = {
+        Math_ApproxF(&this->actor.scale.y, 0.0001f, 0.00015f);
+    }
+    this->actor.scale.z = this->actor.scale.x;
+    Actor_MoveForward(&this->actor);
+    func_8002E4B4(globalCtx, &this->actor, 10.0f, this->actor.scale.x * 3500.0f, 0.0f, 4);
+    if (this->timer < 0x19) {
+        this->alpha -= 0xA;
+        this->alpha = this->alpha < 0 ? 0 : (this->alpha >= 0x100 ? 0xFF : this->alpha);
+    }
+
+    if (this->alpha >= 0x65) {
+        if (this->timer < 0x28) {
+            Collider_CylinderUpdate(&this->actor, &this->collider);
+            this->collider.dim.radius = this->actor.scale.x * 6000.0f;
+            this->collider.dim.height = this->actor.scale.y * 8000.0f;
+            CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+        }
+    }
+    if (this->timer == 0x2E) {
+        s32 i;
+        for (i = 0; i != 0xFFFA; i += 0x1999) {
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ICE_HONO, this->actor.posRot.pos.x,
+                        this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0,
+                        ((s32)(Math_Rand_ZeroOne() * 1000.0f) + i) - 0x1F4, 0, 2);
+        }
+    }
+
+    if (this->timer <= 0) {
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnIceHono_SetupActionSmallFlame(EnIceHono* this) {
+    f32 temp_f0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ice_Hono/func_80A730BC.s")
+    this->actionFunc = EnIceHono_SmallFlameMove;
+    this->timer = 0x2C;
+    this->alpha = 0xFF;
+    if (this->actor.params == 1) {
+        this->smallFlameTargetYScale = (Math_Rand_ZeroOne() * 0.005f) + 0.004f;
+        this->actor.speedXZ = (Math_Rand_ZeroOne() * 1.6f) + 0.5f;
+        return;
+    }
+    this->smallFlameTargetYScale = (Math_Rand_ZeroOne() * 0.005f) + 0.003f;
+    temp_f0 = Math_Rand_ZeroOne();
+    this->actor.speedXZ = (temp_f0 + temp_f0) + 0.5f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ice_Hono/func_80A731C0.s")
+void EnIceHono_SmallFlameMove(EnIceHono* this, GlobalContext* globalCtx) {
+    f32 unk = 10.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ice_Hono/EnIceHono_Init.s")
+    if (this->timer >= 0x15) {
+        Math_ApproxF(&this->actor.scale.x, 0.006f, 0.00016f);
+        Math_ApproxF(&this->actor.scale.y, this->smallFlameTargetYScale * 0.667f, 0.00014f);
+    } else {
+        Math_ApproxF(&this->actor.scale.x, 0.0001f, 0.00015f);
+        Math_ApproxF(&this->actor.scale.y, 0.0001f, 0.00015f);
+    }
+    this->actor.scale.z = this->actor.scale.x;
+    Math_ApproxF(&this->actor.speedXZ, 0, 0.06f);
+    Actor_MoveForward(&this->actor);
+    func_8002E4B4(globalCtx, &this->actor, unk, unk, 0.0f, 5);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ice_Hono/EnIceHono_Destroy.s")
+    if (this->timer < 0x19) {
+        this->alpha -= 0xA;
+        this->alpha = this->alpha < 0 ? 0 : (this->alpha >= 0x100 ? 0xFF : this->alpha);
```

Comment:
I would try the CLAMP macro here. These multiple ternaries on one line looks kinda yucky